### PR TITLE
Grunt default revamp.

### LINKED ISSUE #12325: `bundle exec grunt` doesn't compile builder assets
I was trying to run `grunt` and it didn't compile the builder assets. I talked with @ivanmalagon and told me to run `grunt dev`, which indeed works. We also discovered that you can do `grunt build` to achieve the same, without the watchify.

In my opinion, we have two options here:
1. Change the default grunt task (what you get with running `grunt` without arguments) to do the same as `grunt build` (compile everything)
2. Change the documentation. We should change the installation guide, contributing, etc. to tell everyone to run `grunt build` instead of `grunt`.

I'd prefer 1, as IMHO, the default task should be something useful, not compile half the assets.

cc @xavijam

### PR DESCRIPTION
**Pull request by nobuti**
This PR fixes #12325.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -320,24 +320,30 @@ module.exports = function (grunt) {
...
 
-  grunt.registerTask('default', [
+  grunt.registerTask('pre', [
     'beforeDefault',
     'js',
     'css',
     'manifest'
   ]);
 
+  registerCmdTask('npm-dev', {cmd: 'npm', args: ['run', 'dev']});
```
Comment (xavijam) [inline Gruntfile.js]: shouldn't I be able to run `npm dev` in the console?

**Comment by nobuti** [comment]
npm run dev should work, yes. But that only compiles Builder's js.

**Comment by xavijam** [comment]
:+1:, let's go then man!

**Comment by xavijam** [comment]
cc @AleGoiko and @javitonino ❤️  thanks to @nobuti

**Pull request merged**